app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt

```diff
@@ -145,23 +146,46 @@ class BackgroundService(
 		val parentBackdropUrls = baseItem.parentBackdropImageTags.getUrls(baseItem.parentBackdropItemId)
 		val backdropUrls = itemBackdropUrls.union(parentBackdropUrls)
 
+		loadBackgrounds(backdropUrls)
+	}
+
+	/**
+	 * Use one available backdrop from [SearchHint] as background.
+	 */
+	fun setBackground(searchHint: SearchHint?) {
+		// Check if item is set and backgrounds are enabled
+		if (searchHint == null || !userPreferences[UserPreferences.backdropEnabled])
+			return clearBackgrounds()
```

Comment:
Move this part (from both functions) to the `loadBackgrounds` function to prevent duplicate code.

Reply:
I need some more explanation on this. The load backgrounds function is supposed to be generic and not be aware of whether we have a BaseItem or SearchHint type. This change would make the loadBackground function aware. If this change is really wanted I would suggest maybe just moving everything back into setBackground with a any param and check if that param is not null and is searchhint or baseitem.

`/**
	 * Use all available backdrops from [baseItem] or [SearchHint] as background.
	 */
	fun setBackground(baseItem: Any?) {
		// Check if item is set and backgrounds are enabled
		if (baseItem == null || !userPreferences[UserPreferences.backdropEnabled])
			return clearBackgrounds()

		val backdropUrls: Set<String>

		when (baseItem) {
			is BaseItemDto -> {
				// Get all backdrop urls
				val itemBackdropUrls = baseItem.backdropImageTags.getUrls(baseItem.id)
				val parentBackdropUrls = baseItem.parentBackdropImageTags.getUrls(baseItem.parentBackdropItemId)
				backdropUrls = itemBackdropUrls.union(parentBackdropUrls)
			}
			is SearchHint -> {
				val options = ImageOptions()
				options.imageType = ImageType.Backdrop
				options.tag = baseItem.backdropImageTag

				backdropUrls = setOfNotNull(baseItem.backdropImageItemId?.let { apiClient.GetImageUrl(it, options) })
			}
			else -> {
				return clearBackgrounds();
			}
		}

		loadBackgrounds(backdropUrls)
	}`

something like this maybe?

Reply:
Ignore my message, the parameter is different in both functions so the code is not duplicated. I was only looking at the preferences part.